internal/provider/resource_network.go

```diff
@@ -128,12 +153,36 @@ type NetworkDeployer struct {
 
 func NewNetworkDeployer(ctx context.Context, d *schema.ResourceData, apiClient *apiClient) (NetworkDeployer, error) {
 	var err error
-	nodesIf := d.Get("nodes").([]interface{})
-	nodes := make([]uint32, len(nodesIf))
-	for idx, n := range nodesIf {
-		nodes[idx] = uint32(n.(int))
+	capacityIDsIf := d.Get("capacity_ids").([]interface{})
+	capacityIDs := make([]uint64, len(capacityIDsIf))
+	for idx, c := range capacityIDsIf {
+		capacityIDs[idx] = c.(uint64)
+	}
+
+	capacityDeploymentIDIf := d.Get("capacity_deployment_id").(map[string]interface{})
+	capacityDeploymentID := map[uint64]uint64{}
+	for capacityStr, deploymentID := range capacityDeploymentIDIf {
+		capacityID, err := strconv.ParseUint(capacityStr, 10, 64)
+		if err != nil {
+			return NetworkDeployer{}, errors.Wrap(err, "couldn't parse capacity id")
+		}
+		capacityDeploymentID[capacityID] = deploymentID.(uint64)
 	}
 
+	nodeCapacityIDIf := d.Get("node_capacity_id").(map[string]interface{})
+	nodeCapacity := map[uint32]uint64{}
+	for node, id := range nodeCapacityIDIf {
+		nodeInt, err := strconv.ParseUint(node, 10, 32)
+		if err != nil {
+			return NetworkDeployer{}, errors.Wrap(err, "couldn't parse node id")
```

Comment:
please add the original value that caused the error of parsing to the error message